sdks/wasm/Mono.WebAssembly.DebuggerProxy/FirefoxMonoProxy.cs

```diff
@@ -0,0 +1,510 @@
+
+			var expression = string.Format ("MONO.mono_wasm_set_breakpoint(\"{0}\", {1}, {2})", asm_name, method_token, il_offset);
+			SendCommandToBrowser (actorName, "evaluateJS", expression, Browser, ide);
+		}
+
+		private  void SendBreakpointPositionsCompressed (string script_id, NetworkStream Browser, NetworkStream ide, string line)
+		{
+			string msg = ":{ \"positions\":{ \"" + line + "\":[0]},\"from\":\"" + script_id + "\"}";
+			msg = (msg.Length - 1).ToString () + msg;
+			//System.Console.WriteLine (msg);
+			ide.Write (Encoding.ASCII.GetBytes (msg), 0, msg.Length);
+		}
+		private  void SendBreakableLinesToIde (string script_id, NetworkStream Browser, NetworkStream ide)
+		{
+			Byte [] bytes = new Byte [100000];
+			var id = new SourceId (script_id);
+			var doc = store.GetFileById (id);
+			string breakableLines = ":{\"lines\":[";
+			bool isFirst = true;
+			foreach (var m in doc.Methods) {
+
+				foreach (var sp in m.methodDef.DebugInformation.SequencePoints) {
+					if (sp.IsHidden)
+						continue;
+					if (!isFirst) {
+						breakableLines += ",";
+					} else
+						isFirst = false;
+					breakableLines += sp.StartLine.ToString ();
+				}
+			}
+		Next:
+			breakableLines += "],\"from\":\"";
+			breakableLines += script_id;
+			breakableLines += "\"}";
+			var msg = (breakableLines.Length - 1).ToString () + breakableLines;
+			ide.Write (Encoding.ASCII.GetBytes (msg), 0, msg.Length);
+			//var bytesRec = ide.Read(bytes, 0, bytes.Length);
+			//System.Console.WriteLine("O que achou do source:" + Encoding.ASCII.GetString(bytes, 0, bytesRec));
+			//107:{"lines":[15,16,17,19,20,22,23,25,27,29,30,31,32,33,35,36,38,39],"from":"server1.conn155.child28/source26"}
+		}
+		private  async Task SendFileToIde (string script_id, NetworkStream Browser, NetworkStream ide)
+		{
+			var id = new SourceId (script_id);
+			var src_file = store.GetFileById (id);
+
+			var res = new StringWriter ();
+			//res.WriteLine ($"//{id}");
+
+			try {
+				var uri = new Uri (src_file.Url);
+				if (uri.IsFile && File.Exists (uri.LocalPath)) {
+					using (var f = new StreamReader (File.Open (src_file.SourceUri.LocalPath, FileMode.Open))) {
+						await res.WriteAsync (await f.ReadToEndAsync ());
+					}
+					//
+					var o = JObject.FromObject (new {
+						source = res.ToString (),
+						contentType = "text/javascript",
+						from = script_id
+					});
+					var str2 = o.ToString ();
+					var msg = str2.Length.ToString ();
+					msg += ":";
+					msg += str2;
+					ide.Write (Encoding.ASCII.GetBytes (msg), 0, msg.Length);
+				} else if (src_file.SourceLinkUri != null) {
+					var doc = await new WebClient ().DownloadStringTaskAsync (src_file.SourceLinkUri);
+					await res.WriteAsync (doc);
+
+					var o = JObject.FromObject (new {
+						source = res.ToString (),
+						contentType = "text/javascript",
+						from = script_id
+					});
+
+					var str2 = o.ToString ();
+					var msg = str2.Length.ToString ();
```

Comment:
Here and elsewhere: can use `var msg = String.Format ($"{str2.Length}:{str2}");`

Reply:
I guess you mean `String.Format` or `$"{...}"`. Combining them doesn't make sense :-)